proto/amino/amino.proto

```diff
@@ -76,4 +76,10 @@ extend google.protobuf.FieldOptions {
   // out := AminoJSONEncoder(&f);
   // out == {"baz":""}
   bool dont_omitempty = 11110005;
+
```

Comment:
These options were added to achieve marshal parity for messages with a `oneof` field, e.g. StakeAuthorization: https://github.com/cosmos/cosmos-sdk/blob/594632037d88e849f1e6f718a195ed159501cb18/proto/cosmos/staking/v1beta1/authz.proto#L22-L29

Since the legacy encoder uses golang reflection it produces JSON shaped like below.
```
{"type":"cosmos-sdk/StakeAuthorization","value":{"Validators":{"type":"cosmos-sdk/StakeAuthorization/AllowList","value":{"allow_list":{"address":["foo"]}}}}}
```
The protoreflect default field traversal differs:
```
{"type":"cosmos-sdk/StakeAuthorization","value":{"allow_list":{"address":["foo"]}}}
```
The options are introspected and used in the encoder.  https://github.com/cosmos/cosmos-sdk/blob/732bcafde4b280d8516ddc8abaad2ad04f0ddb58/x/tx/aminojson/json_marshal.go#L160-L162